src/core/src/bootstrap/EnvLayer.py

```diff
@@ -46,6 +47,15 @@ def __init__(self):
     def is_distro_azure_linux(distro_name):
         return any(x in distro_name for x in Constants.AZURE_LINUX)
 
+    @staticmethod
+    def is_distro_azure_linux_3_or_beyond():
+        # type: () -> bool
+        """ Checks if the current distro is Azure Linux 3 """
+
+        version = distro.os_release_attr('version')
+        major = version.split('.')[0] if version else None
+        return major is not None and int(major) >= 3
```

Comment:
Function body contents only do a part of what the function name promises outwards?

Basically the major version of whatever OS the system may be running is coming back as >=3. This could be anything right, outside the specific way it's used (which is incidental)?

And if the code doesn't work in some case on a matching AzLinux system, it says' No'?

Reply:
Added an Azure Linux check before verifying the major version